### PR TITLE
[batch] duration includes all job tasks

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: batch/batch/server/server.py [hunk 1]
```diff
@@ -483,7 +483,7 @@ def is_successful(self):
             terminated = pod.status.container_statuses[0].state.terminated
             self.exit_code = terminated.exit_code
             if terminated.finished_at is not None and terminated.started_at is not None:
-                self.duration = (terminated.finished_at - terminated.started_at).total_seconds()
+                self.duration += (terminated.finished_at - terminated.started_at).total_seconds()
             else:
```
Comment (danking) [inline batch/batch/server/server.py]: won't this fail since duration starts as None?

Comment (jigold) [reply batch/batch/server/server.py]: fixed

**Comment by jigold** [comment]
How do we want to handle the case where one of the tasks does not have timing information?

```
                log.warning(f'job {self.id} has pod {pod.metadata.name} which is '
                            f'terminated but has no timing information. {pod}')
```

I think it should be None for all tasks, which is not the behavior I have now.

**Comment by jigold** [comment]
I changed the behavior of duration to be None if one of the job tasks has no timing data.

**Comment by jigold** [comment]
It's already in there. Look 2 lines below the changes. Can't figure out how to get a link to that line...

**Pull request merged**